merry holidays to those who celebrate

pillowtrucker opened the pull request:
I recently had to fix the same thing with another wgpu/winit thing so I figured might as well do this too

pillowtrucker committed:
update deps

pillowtrucker committed:
ignore clippy

pillowtrucker committed:
don't forget lovely wasm target

Speykious commented on examples/render-opengl/src/opengl.rs:
```diff
@@ -40,12 +41,18 @@ pub fn launch_opengl_window() -> Result<App, Box<dyn Error>> {
-				.max_by_key(|c| c.num_samples())
+				.filter(|c| match c {
+					Config::Egl(c) => c.srgb_capable(),
+					_ => false,
```
This should be `true` in the default case, otherwise it'll filter everything out and get the following error:
```
thread 'main' panicked at examples/render-opengl/src/opengl.rs:56:18:
called `Option::unwrap()` on a `None` value
```

Speykious reviewed (changes requested):
The WGPU example doesn't work right now and it seems like dependencies weren't all updated. I'm gonna try to see what I can do.

In any case thank you so much :D

pillowtrucker committed:
whoops don't mess up Glx

pillowtrucker commented:
hmm wgpu definitely both builds and works fine for me, not sure what could be wrong there. wasm32 target might not work fully as it's painful to test, so my only test for that is whether it compiles

Speykious committed:
Fix WGPU example after update

winit explicitly needs the `rwh_05` feature if we have to use `raw-window-handle` v0.5.

Speykious commented:
The `rwh_05` feature was missing

Speykious commented:
> wasm32 target might not work fully as it's painful to test, so my only test for that is whether it compiles

`trunk serve` doesn't work for you?

Speykious commented on examples/render-opengl/src/opengl.rs:
```diff
@@ -40,12 +41,18 @@ pub fn launch_opengl_window() -> Result<App, Box<dyn Error>> {
 		.build(&events, <_>::default(), |configs| {
 			configs
-				.filter(|c| c.srgb_capable())
-				.max_by_key(|c| c.num_samples())
+				.filter(|c| match c {
+					Config::Egl(c) => c.srgb_capable(),
+					Config::Glx(c) => c.srgb_capable(),
+				})
+				.max_by_key(|c| match c {
+					Config::Egl(c) => c.num_samples(),
+					Config::Glx(c) => c.num_samples(),
+				})
```
So why this change? I just tried and `c.srgb_capable()` is still available on the root data structure. Same for `c.num_samples()`. Now it breaks on Mac and Windows.

pillowtrucker replied:
not in my version. I feel like we're testing on different libraries or compilers. I'm fixing up the other platforms rn

pillowtrucker replied:
I think I had the wrong version of the import

Speykious replied:
Ah, turns out it was a wrong import, alright :v

pillowtrucker committed:
no Glx on Win/Mac

pillowtrucker committed:
brain fart, wrong import confused RA for me

Speykious committed:
Remove unused import

Speykious commented:
Merry Christmas :D
